primitives/numeric/src/lib.rs

```diff
@@ -72,6 +73,8 @@ pub enum NumericError {
     Negative,
     /// Malformed: expecting number with optional decimal point (10, 10.02)
     Malformed,
+    /// Represents a failure to convert to/from [`Numeric`] to the specified type
+    Conversion,
```

Comment:
I think it's better to use different error types for different code paths.
In such case we don't have to match against error not possible in given context.

Reply:
I made it into a separate `TryFromNumericError`